pkg/objectio/column.go

```diff
@@ -0,0 +1,168 @@
+// Copyright 2021 Matrix Origin
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package objectio
+
+import (
+	"bytes"
+	"context"
+	"encoding/binary"
+	"github.com/matrixorigin/matrixone/pkg/fileservice"
+	"github.com/matrixorigin/matrixone/pkg/vm/engine/tae/index"
+)
+
+type ColumnBlock struct {
+	meta  *ColumnMeta
+	block *Block
+}
+
+func NewColumnBlock(idx uint16, block *Block) ColumnObject {
+	meta := &ColumnMeta{
+		idx:         idx,
+		zoneMap:     &index.ZoneMap{},
+		bloomFilter: Extent{},
+	}
+	col := &ColumnBlock{
+		block: block,
+		meta:  meta,
+	}
+	return col
+}
+
+func (cb *ColumnBlock) GetData() (*fileservice.IOVector, error) {
+	var err error
+	data := &fileservice.IOVector{
+		FilePath: cb.block.object.name,
+		Entries:  make([]fileservice.IOEntry, 1),
+	}
+	data.Entries[0] = fileservice.IOEntry{
+		Offset: int(cb.meta.location.Offset()),
+		Size:   int(cb.meta.location.Length()),
+	}
+	err = cb.block.object.fs.Read(context.Background(), data)
+	if err != nil {
+		return nil, err
+	}
+	return data, nil
+}
+
+func (cb *ColumnBlock) GetIndex() (*fileservice.IOVector, error) {
+	var err error
+	data := &fileservice.IOVector{
+		FilePath: cb.block.object.name,
+		Entries:  make([]fileservice.IOEntry, 1),
+	}
+	data.Entries[0] = fileservice.IOEntry{
+		Offset: int(cb.meta.bloomFilter.Offset()),
```

Comment:
IOEntry shoulld not use int.  Always use int32, int64 etc,

Reply:
IOEntry is a structure in fileservice and needs to be modified by fileservice.

Reply:
I will change it to int64 in fileservice.